system_updates.py

```
@@ -0,0 +1,705 @@
+            raise ValueError("Livepatch token is required.")
+        distro = self.get_distribution_info()
+        if distro["id"] != "ubuntu":
+            raise RuntimeError("Ubuntu Livepatch can only be set up on Ubuntu.")
+        if self.runtime.is_fake:
+            return self.get_livepatch_status()
+
+        binary = shutil.which("canonical-livepatch")
+        if not binary:
+            if not shutil.which("snap"):
+                raise RuntimeError("snap is required to install canonical-livepatch.")
+            subprocess.run(["sudo", "snap", "install", "canonical-livepatch"], check=True, capture_output=True, text=True)
```

Ruff (0.15.11)

[error] Starting a process with a partial executable path

(S607)